In [8]:
import pandas as pd
import os
import urllib.request

In [2]:
train_df = pd.read_csv(r"C:\Users\lbenzemma\Desktop\Projets Master2 MOSEF\deep learning\projet DL\DL\data\train.csv")  
category_df = pd.read_csv(r"C:\Users\lbenzemma\Desktop\Projets Master2 MOSEF\deep learning\projet DL\DL\data\train_label_to_category.csv")  
print("Train.csv :")
print(train_df.head())
print("\ntrain_label_to_category.csv :")
print(category_df.head())
print("\nCategories uniques :")
print(category_df["category"].unique())

Train.csv :
                 id                                                url  \
0  6e158a47eb2ca3f6  https://upload.wikimedia.org/wikipedia/commons...   
1  202cd79556f30760  http://upload.wikimedia.org/wikipedia/commons/...   
2  3ad87684c99c06e1  http://upload.wikimedia.org/wikipedia/commons/...   
3  e7f70e9c61e66af3  https://upload.wikimedia.org/wikipedia/commons...   
4  4072182eddd0100e  https://upload.wikimedia.org/wikipedia/commons...   

   landmark_id  
0       142820  
1       104169  
2        37914  
3       102140  
4         2474  

train_label_to_category.csv :
   landmark_id                                           category
0            0  http://commons.wikimedia.org/wiki/Category:Hap...
1            1  http://commons.wikimedia.org/wiki/Category:Lui...
2            2  http://commons.wikimedia.org/wiki/Category:Gra...
3            3  http://commons.wikimedia.org/wiki/Category:Twe...
4            4  http://commons.wikimedia.org/wiki/Category:San...

Categories un

**Filtrer les landmarks correspondant aux pays de l'Océanie**

In [12]:
pays_oceanie = ["Australia", "Fiji", "Kiribati", "Marshall Islands", "Micronesia", 
                "Nauru", "New Zealand", "Palaos", "Papua New Guinea", "Samoa", 
                "Tonga", "Tuvalu", "Vanuatu"]

oceanie_landmarks = category_df[category_df["category"].str.contains('|'.join(pays_oceanie), case=False, na=False)]

landmarks_oceanie_ids = oceanie_landmarks["landmark_id"].unique()

oceanie_landmarks.to_csv("data\landmarks_oceanie.csv", index=False)

print(f"Nombre de landmarks en Océanie : {len(landmarks_oceanie_ids)}")


Nombre de landmarks en Océanie : 172


In [13]:
landmarks_par_pays = {}
for pays in pays_oceanie:
    count = oceanie_landmarks[oceanie_landmarks["category"].str.contains(pays, case=False, na=False)]["landmark_id"].nunique()
    landmarks_par_pays[pays] = count
for pays, count in landmarks_par_pays.items():
    print(f"{pays} : {count} landmarks")

Australia : 151 landmarks
Fiji : 4 landmarks
Kiribati : 0 landmarks
Marshall Islands : 0 landmarks
Micronesia : 0 landmarks
Nauru : 2 landmarks
New Zealand : 0 landmarks
Palaos : 0 landmarks
Papua New Guinea : 0 landmarks
Samoa : 3 landmarks
Tonga : 12 landmarks
Tuvalu : 0 landmarks
Vanuatu : 0 landmarks


**L'Australie domine avec 151 landmarks, ce qui est attendu vu la taille du pays.**
**Palau arrive en deuxième avec 85 landmarks**
**Tonga, Fidji, Samoa et Nauru ont quelques landmarks**
**Aucun landmark trouvé pour la Nouvelle-Zélande, la Papouasie-NG, Tuvalu, Kiribati, Micronésie, Îles Marshall et Vanuatu.**

In [14]:
pays_disponibles = ["Australia", "Palau", "Tonga", "Fiji", "Samoa", "Nauru"]

landmarks_filtres = oceanie_landmarks[oceanie_landmarks["category"].str.contains('|'.join(pays_disponibles), case=False, na=False)]
landmarks_ids = landmarks_filtres["landmark_id"].unique()

train_oceanie = train_df[train_df["landmark_id"].isin(landmarks_ids)]

train_oceanie.to_csv(r"data\train_oceanie_final.csv", index=False)

print(f"Nombre d'images à télécharger : {len(train_oceanie)}")


Nombre d'images à télécharger : 3050
